### ISSUE TITLE
Entering unmatched text collapses the control.

### ISSUE THREAD

**Issue by mendeza** [mtkopone/zelect#7]
![zelect-collapsed](https://f.cloud.github.com/assets/1161280/505143/68fe7930-bd11-11e2-9bb3-fa211165d7fb.png)

**Comment by mtkopone** [comment]
I'm sorry, but I'm going to need some more info to know what this is about?

**Comment by mtkopone** [comment]
This might be a browser issue? Which one we're you using?

Here's chrome: 

![screen shot 2013-05-15 at 9 27 05 am](https://f.cloud.github.com/assets/285399/505473/8baac74c-bd28-11e2-92eb-8390efc1f6de.png)

**Comment by ErtugKaya** [comment]
Just press enter after that. It collapses for me too. Using latest Chrome.

Edit: It collapses because text inside div is gone.

**Comment by brianmhunt** [comment]
I can confirm that this behaviour occurs on Chrome on Mac, Chrome on iOS, and Safari on iOS.

**Comment by brianmhunt** [comment]
A selection of the trace of the issue when it exhibits is as follows:
- Line 54 - listNavigator.current() returns []
- Line 290 - `$list.find(".current")` returns []
- Line 54 - listNavigator.curent().data('.zelect-item') returns `undefined`
- Line 84 - `selectItem(undefined, undefined)`
- Line 129 - `content` is `""`; The call `$obj.text('')`

It seems that the issue is the `div` collapsing because the text is set to empty (''). 

There are three apparent workarounds: 1. putting a `&nbsp;` in the `div` whenever the content is blank, or 2. manually setting the height, or 3. adding a character in a `<span>` whose visibility is hidden to the div.

An example of fix 1 (probably the easiest) would be changing [`renderContext`](https://github.com/mtkopone/zelect/blob/master/zelect.js#L128) to:

``` javascript
$obj[htmlOrText(content)](content === '' ? "&nbsp;" : content);
return $obj;
function htmlOrText(x) {
      return (x instanceof jQuery || x.nodeType != null || x === '')
              ? 'html' : 'text'
}
```

This ensures that `$obj.html("&nbsp;")` is called when the content is the empty string.

There may be other and better workarounds, but this'll probably do the trick.

**Comment by mtkopone** [comment]
This should be fixed in 0.0.8.

Pressing enter really shouldn't have "made a selection" when there is no selection to be made... So I fixed this by doing nothing when listNavigator.current() doesn't hit anything...
